[Bug Report] Explicitly defined theme color for a component reverts back on page navigation even though theme property changed

### Versions and Environment
**Vuetify:** 2.0.0-beta.8
**Vue:** 2.6.10
**Browsers:** Vivaldi 2.7.1609.4
**OS:** Windows 10

### Steps to reproduce
1.  Run the Python script `scripts/create_blog_metadata.py`. The prerequisite python packages can be installed using the `scripts/python-requirements.txt` file.
2. Run `npm run nuxt-dev`
3. Open the website (localhost) in any browser.
4. Click on the `change theme` icon in the top nav bar. Your theme will now change from the default dark to light.
5. Click on a link (any link, say `Legal` in the top bav bar).
6. The top nav bar and footer will now revert back to the dark theme colors.
7. Go back to the home page. You will see that all components that have explicitly a color defined have reverted back to the default dark color.

### Expected Behavior
Colors as defined under theme options stay put after navigation to a different link, even though the theme property has changed from dark to light and vice versa.

### Actual Behavior
All components that have explicitly a color defined have reverted back to the default dark color.

### Reproduction Link
<a href="https://github.com/manastalukdar/manastalukdar.github.io/tree/source/website" target="_blank">https://github.com/manastalukdar/manastalukdar.github.io/tree/source/website</a>

### Other

- Using Nuxt.
- Using the `@nuxtjs/vuetify`, `1.0.0-beta.7` module.


<!-- generated by vuetify-issue-helper. DO NOT REMOVE -->

@manastalukdar I followed your instructions, it ends up with
```
Cannot find module 'src/components/pages/index.js' from '/home/kevin/tmp/manastalukdar.github.io/website'
```
on my computer :/

Even if your custom SCSS variables will be broken, can you try setting `treeShake: false` ? Just to see if it's related. Thanks.

--- a/packages/vuetify/src/services/theme/__tests__/theme.spec.ts
+++ b/packages/vuetify/src/services/theme/__tests__/theme.spec.ts
@@ -183,17 +183,9 @@ describe('Theme.ts', () => {
     expect(ssrContext.head).toMatchSnapshot()
   })
 
-  it('should add fake child element for vue-meta with ssr support', () => {
+  it('should add fake child element for vue-meta support', () => {
     const theme = new Theme(mock)
     ;(instance as any).$meta = {}
-
-    // $isServer is read only but need to be set for test purpose : https://github.com/vuejs/vue/issues/9232#issuecomment-477914392
-    Object.setPrototypeOf(
-      instance,
-      new Proxy(Object.getPrototypeOf(instance), {
-        get: (target, key, receiver) => key === '$isServer' ? true : Reflect.get(target, key, receiver),
-      })
-    )
 
     expect(instance.$children).toHaveLength(0)
 
@@ -205,7 +197,7 @@ describe('Theme.ts', () => {
     const head = options.head
 
     expect(head).toBeTruthy()
-    expect(head).toMatchSnapshot()
+    expect(head()).toMatchSnapshot()
   })
 
   it('should react to theme changes', async () => {

--- a/packages/vuetify/src/services/theme/index.ts
+++ b/packages/vuetify/src/services/theme/index.ts
@@ -110,10 +110,10 @@ export class Theme extends Service {
     if (this.disabled) return
 
     /* istanbul ignore else */
-    if (ssrContext) {
-      this.initSSR(ssrContext)
-    } else if (root.$isServer && (root as any).$meta) {
+    if ((root as any).$meta) {
       this.initVueMeta(root)
+    } else if (ssrContext) {
+      this.initSSR(ssrContext)
     }
 
     this.initTheme()
@@ -178,15 +178,17 @@ export class Theme extends Service {
   private initVueMeta (root: Vue) {
     const options = this.options || {}
     root.$children.push(new Vue({
-      head: {
-        style: [
-          {
-            cssText: this.generatedStyles,
-            type: 'text/css',
-            id: 'vuetify-theme-stylesheet',
-            nonce: options.cspNonce,
-          },
-        ],
+      head: () => {
+        return {
+          style: [
+            {
+              cssText: this.generatedStyles,
+              type: 'text/css',
+              id: 'vuetify-theme-stylesheet',
+              nonce: options.cspNonce,
+            },
+          ],
+        }
       },
     } as any))
   }

